Sources/Datadog/DatadogCore/Context/AppStatePublisher.swift

```diff
@@ -0,0 +1,145 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal final class AppStatePublisher: ContextValuePublisher {
+    typealias Snapshot = AppStateHistory.Snapshot
+
+    /// The default publisher queue.
+    private static let defaultQueue = DispatchQueue(
+        label: "com.datadoghq.app-state-publisher",
+        target: .global(qos: .utility)
+    )
+
+    /// The initial history value.
+    var initialValue: AppStateHistory
```

Comment:
nit/ Seems it can be immutable, no?
```suggestion
    /// The initial history value.
    let initialValue: AppStateHistory
```